operator/v1/types_ingress.go

```diff
@@ -545,6 +545,22 @@ type AWSLoadBalancerParameters struct {
 	//
 	// +optional
 	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+
+	// subnets specifies the list of subnets for the load balancer to
+	// route traffic to. The values can be either a subnet ID or name.
+	// In order for the load balancer to be provisioned with subnets:
+	// * Each subnet must exist.
+	// * Each subnet must be from a different availability zone.
+	// * The load balancer service must be deleted.
+	//
+	// When omitted, the subnets will be auto-discovered per availability zone.
+	//
+	// +optional
+	// +listType=atomic
+	// +openshift:enable:FeatureGate=IngressControllerLBSubnetsAWS
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="subnets cannot contain duplicates"
```

Comment:
Do you know if this validation is case-sensitive?  The subnet names can be, so there could correctly be both subnetB and subnetb in the list.

Reply:
I don't know how much it matters, but I tested this with two empty subnets and that returns false.

Reply:
> Do you know if this validation is case-sensitive? The subnet names can be, so there could correctly be both subnetB and subnetb in the list. You could just add a test to the test yaml.

Good question. It is indeed case-sensitive. Good idea, I will do that.

> I don't know how much it matters, but I tested this with two empty subnets and that returns false.

Oh nice, I didn't know about this CEL playground website.

Good point. The CEL is working as expected with two empty subnets, but an empty length subnet is not a valid subnet reference. Adding `// +kubebuilder:validation:MinLength=1`